schemas/com.snowplowanalytics.snowplow.enrichments/sql_query_enrichment_config/jsonschema/1-0-1

```diff
@@ -0,0 +1,214 @@
+											"type": "string",
+											"minLength": 1
+										},
+										"password": {
+											"type": "string"
+										},
+										"database": {
+											"type": "string",
+											"minLength": 1
+										}
+									},
+									"required": ["host", "port", "sslMode", "username", "password", "database"],
+									"additionalProperties": false
+								}
+							},
+							"required": ["postgresql"],
+							"additionalProperties": false
+						},
+
+						{
+							"type": "object",
+							"properties": {
+								"mysql": {
+									"type": "object",
+									"properties": {
+										"host": {
+											"type": "string"
+										},
+										"port": {
+											"type": "integer",
+											"minimum": 1,
+											"maximum": 65535
+										},
+										"sslMode": {
+											"type": "boolean"
+										},
+										"username": {
+											"type": "string",
+											"minLength": 1
+										},
+										"password": {
+											"type": "string"
+										},
+										"database": {
+											"type": "string",
+											"minLength": 1
+										}
+									},
+									"required": ["host", "port", "sslMode", "username", "password", "database"],
+									"additionalProperties": false
+								}
+							},
+							"required": ["mysql"]
+						}
+					],
+					"additionalProperties": true
+				},
+				"query": {
+					"type": "object",
+					"properties": {
+						"sql": {
+							"type": "string"
+						}
+					},
+					"required": ["sql"],
+					"additionalProperties": false
+				},
+				"output": {
+					"type": "object",
+					"properties": {
+						"expectedRows": {
+							"type": "string",
+							"enum": ["AT_LEAST_ONE", "AT_LEAST_ZERO", "AT_MOST_ONE", "EXACTLY_ONE"]
+						},
+						"json": {
+							"type": "object",
+							"properties": {
+								"schema": {
+									"type": "string",
+									"pattern": "^iglu:([a-zA-Z0-9-_.]+)/([a-zA-Z0-9-_]+)/([a-zA-Z0-9-_]+)/([1-9][0-9]*(?:-(?:0|[1-9][0-9]*)){2})$"
+								},
+								"propertyNames": {
+									"type": "string",
+									"enum": ["AS_IS", "CAMEL_CASE", "PASCAL_CASE", "SNAKE_CASE", "LOWER_CASE", "UPPER_CASE"]
+								},
+								"describes": {
+									"type": "string",
+									"enum": ["ALL_ROWS", "EVERY_ROW"]
+								}
+							},
+							"required": ["schema", "propertyNames", "describes"],
+							"additionalProperties": false
+						}
+					},
+					"additionalProperties": false
+				},
+				"cache": {
+					"type": "object",
+					"properties": {
+						"size": {
+							"type": "integer",
+							"minimum": 0
+						},
+						"ttl": {
+							"type": "integer",
+							"minimum": 0,
+							"maximum": 86400
+						}
+					},
+					"additionalProperties": false,
+					"required": ["size", "ttl"]
+				},
+				"ignoreOnError": {
+					"type": "boolean"
+				}
+			},
+			"additionalProperties": false,
+			"required": ["inputs", "database", "query", "output", "cache", "ignoreOnError"]
```

Comment:
The Redshift loader will now handle this gracefully, so technically there is nothing to worry about.

But....

You are correct, we shouldn't add a new field as a required field.  Because it breaks the principles of using schema ver.

The guiding principle of schema versioning is: The latest version of a schema _must_ describe all historic data.  In this case, json that was valid against the 1-0-0 schema is not valid against the 1-0-1 schema, because it would be missing the `ignoreOnError` field.

I think it would be better to make `ignoreOnError` an optional field.  And I don't think that causes any problems for how we want to use this schema.

cc @pondzix

Reply:
The same issues is in the API enrichment bump also!

Reply:
Ok, I made `ignorOnError` optional for both: SQL and API enrichments